unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+
+    def test_function_check_buffer_message_matching_command_and_id_fixture_call_success(self,
+                                                                                        check_buffer_message_fixture):
+        check_buffer_message_fixture._check_buffer_message_matching_command_and_id()
+        check_buffer_message_fixture._check_buffer_message_matching_command_and_id.assert_called_once()
+
+    def test_function_register_call_success(self, register_fixture):
+        register_fixture.register()
+        register_fixture.register.assert_called_once()
+
+    def test_function_unregister_call_success(self, unregister_fixture):
+        unregister_fixture.unregister()
+        unregister_fixture.unregister.assert_called_once()
+
+    def test_function_on_register_call_success(self, on_register_fixture):
+        on_register_fixture._on_register()
+        on_register_fixture._on_register.assert_called_once()
+
+    def test_function__check_transaction_id_cached_out_call_success(self,
+                                                                    check_transaction_id_cached_out_call_success_fixture):
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out()
+        check_transaction_id_cached_out_call_success_fixture._check_transaction_id_cached_out.assert_called_once()
+
+    def test_check_buffer_message_matching_command_and_id_throws_exception(self):
+        message = {}
+        with pytest.raises(RedisAPIException,
+                           match='The answer message does not contain a valid '
+                                 '\'transaction_id\' member.') as ex:
+            redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
+            redis_client_base._check_buffer_message_matching_command_and_id(message)
+
+    def test_register_blocking_true_throws_exception(self):
+        with pytest.raises(RedisAPIException,
+                           match='API registration process timed out. Server will continue processing '
+                                 'your request on the background and will notify you as soon as the '
+                                 'registration has been completed.') as ex:
+            redis_client_base = RedisClientBase(area_id=area_id, autoregister=True)
+            redis_client_base.register(is_blocking=True)
+
+    def test_register_blocking_false_published_success(self):
+        # arrange
+        redis_client_base = RedisClientBase(area_id=area_id, autoregister=False)
+        redis_client_base.register(is_blocking=False)
+        assert redis_client_base._blocking_command_responses["register"]["name"] == area_id
+
+    def test_register_blocking_false_throw_exception(self, redis_client_auto_register):
+        with pytest.raises(RedisAPIException,
+                           match='API is already registered to the market.') as ex:
+            self.redis_client_base = redis_client_auto_register
```

Comment:
Why are you using an instance variable now? I'm a bit confused by the fact that you alternate the following styles in your tests:
- instance variable `self.obj`
- instance variable `self.redis_client_base` (instead of directly using the fixture)
- local variable `redis_client`

I think that local variables should be enough, but in any case it would be good to have just one approach if possible.

Reply:
Yes, you are right. We can directly use the fixture. I will adapt the suggested changes by using fixtures.